modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/subscription/SubscriptionManagerImpl.kt

```diff
@@ -116,34 +115,35 @@ class SubscriptionManagerImpl @Inject constructor(
         }.distinctUntilChanged()
     }
 
-    override fun getSubscriptionStatus(allowCache: Boolean): Single<SubscriptionStatus> {
+    override fun getSubscriptionStatusRxSingle(allowCache: Boolean): Single<SubscriptionStatus> {
+        return rxSingle {
+            getSubscriptionStatus(allowCache)
+        }
+    }
```

Comment:
I have added a Coroutine version of this method below, and I will remove this method from the code slowly.